commit 5353e5ee0063264ee4a557a98a19a9d0a7d2b053
Author: msweet <msweet@a1ca3aef-8c08-0410-bb20-df032aa958be>
Date:   Wed Feb 19 20:15:53 2014 +0000

    The dnssd backend did not always report all discovered printers using Avahi
    (STR #4365)
    
    Keep track of active browsers, don't stop until Avahi gives up.
    
    
    
    git-svn-id: svn+ssh://src.apple.com/svn/cups/cups.org/trunk@11622 a1ca3aef-8c08-0410-bb20-df032aa958be

--- a/backend/dnssd.c
+++ b/backend/dnssd.c
@@ -95,6 +95,7 @@
 static AvahiSimplePoll	*simple_poll = NULL;
 					/* Poll information */
 static int		got_data = 0;	/* Got data from poll? */
+static int		browsers = 0;	/* Number of running browsers */
 #endif /* HAVE_AVAHI */
 
 
@@ -345,6 +346,7 @@
     return (1);
   }
 
+  browsers = 6;
   avahi_service_browser_new(client, AVAHI_IF_UNSPEC,
 			    AVAHI_PROTO_UNSPEC,
 			    "_fax-ipp._tcp", NULL, 0,
@@ -558,7 +560,11 @@
 
       fprintf(stderr, "DEBUG: sent=%d, count=%d\n", sent, count);
 
+#ifdef HAVE_AVAHI
+      if (sent == cupsArrayCount(devices) && browsers == 0)
+#else
       if (sent == cupsArrayCount(devices))
+#endif /* HAVE_AVAHI */
 	break;
     }
   }
@@ -710,9 +716,12 @@
 	break;
 
     case AVAHI_BROWSER_REMOVE:
-    case AVAHI_BROWSER_ALL_FOR_NOW:
     case AVAHI_BROWSER_CACHE_EXHAUSTED:
         break;
+
+    case AVAHI_BROWSER_ALL_FOR_NOW:
+	browsers--;
+	break;
   }
 }
 
